provider/datagen/src/transform/cldr/units/mod.rs

```diff
@@ -2,34 +2,137 @@
+
+                for i in 0..temp_num.len() {
+                    if !contains_alphabetic_chars(temp_num[i].as_str())
+                        || is_scientific_number(temp_num[i].as_str())
+                    {
+                        continue;
+                    }
+
+                    num_of_const_with_text += 1;
+                    if let Some((rnum, rden, rconstant_type)) =
+                        constants_map_in_str_form.get(temp_num[i].as_str())
+                    {
+                        temp_num.remove(i);
+                        // append the elements in rnum to num and rden to den
+                        temp_num.append(&mut rnum.clone());
+                        temp_den.append(&mut rden.clone());
+
+                        if *rconstant_type == ConstantType::Approximate {
+                            temp_constant_type = ConstantType::Approximate;
+                        }
+                    }
+                }
+
+                for i in 0..temp_den.len() {
+                    if !contains_alphabetic_chars(temp_den[i].as_str())
+                        || is_scientific_number(temp_den[i].as_str())
+                    {
+                        continue;
+                    }
+
+                    num_of_const_with_text += 1;
+                    if let Some((rnum, rden, rconstant_type)) =
+                        constants_map_in_str_form.get(temp_den[i].as_str())
+                    {
+                        temp_den.remove(i);
+                        // append the elements in rnum to den and rden to num
+                        temp_num.append(&mut rden.clone());
+                        temp_den.append(&mut rnum.clone());
+
+                        if *rconstant_type == ConstantType::Approximate {
+                            temp_constant_type = ConstantType::Approximate;
+                        }
+                    }
+                }
+
+                constants_with_constants_map_replaceable
+                    .insert(cons_name, (temp_num, temp_den, temp_constant_type));
+            }
+
+            constants_map_in_str_form.clear();
+            constants_map_in_str_form = constants_with_constants_map_replaceable;
+
+            if num_of_const_with_text == 0 {
+                break;
+            }
+        }
+
+        let mut constants_map = BTreeMap::<&str, ConstantValue>::new();
+
+        for (cons_name, (num, den, constant_type)) in constants_map_in_str_form.iter() {
+            let value = convert_array_of_strings_to_fraction(num, den)?;
+            let (num, den, sign, cons_type) =
+                transform_fraction_to_constant_value(value, *constant_type)?;
+            constants_map.insert(
+                cons_name,
+                ConstantValue {
+                    numerator: ZeroVec::from_iter(num),
+                    denominator: ZeroVec::from_iter(den),
+                    sign,
+                    constant_type: cons_type,
+                },
+            );
```

Comment:
nit: this could all be moved into the `map` function below, without the need for another intermediary map

Reply:
I prefer to split them for the clarity of code.

Reply:
A `map` is a lot clearer than declaring a collection and then using a for loop to fill it. It is also more efficient as it can preallocate. You can use two `map`s on the iterator if you want to separate creating the non-ULE type and converting to ULE.

Reply:
done